WeatherService.Tests/BotsTest/WeatherStrategyShould.cs

```diff
@@ -0,0 +1,101 @@
+﻿using FluentAssertions;
+using Moq;
+using System.Collections.Generic;
+using WeatherMonitoringReportingService.Bots;
+using WeatherMonitoringReportingService.Bots.Strategy;
+using WeatherMonitoringReportingService.Data;
+using Xunit;
+
+namespace WeatherService.Tests
+{
+    public class WeatherStrategyShould
+    {
+        [Fact]
+        public void CallParseOnce()
+        {
+            Mock<IWeatherStrategy> mockStrategy = new Mock<IWeatherStrategy>();
+
+            mockStrategy.Object.CheckActivation(new WeatherData(), new ConfigurationData());
+
+            mockStrategy.Verify(
+                strategy => strategy.CheckActivation(It.IsAny<WeatherData>(), It.IsAny<ConfigurationData>()), 
+                Times.Once());
+
+        }
+
+        [Theory]
+        [InlineData(false, 80, 70, false)]
+        [InlineData(false, 70, 80, false)]
+        [InlineData(true, 80, 70, true)]
+        [InlineData(true, 70, 80, false)]
+        public void CheckActivationRainStrategy(bool enabled, double weatherHumidity,
```

Comment:
consider better naming for test cases. this one could be `RainStrategy_ShouldActivateBasedOnHumidityThreshold `